articles/140_topic_and_service_name_mapping.md

```diff
@@ -224,69 +257,73 @@ Here is a non-exhaustive list of prefixes:
+### Changing ROS Names During Runtime
 
-#### DDS Partitions Characteristics
+If the ROS name needs to be changed during runtime (after the publisher or subscriber has already been created and is in use), then some details about DDS partitions might be useful to consider.
 
-A DDS topic at this point is the basename token within a ROS topic.
-The namespace of a ROS topic is implemented within the partition field of DDS.
-DDS partitions are implemented as a string array within the Subscription/Publisher QoS, allowing up to 64 items, where each string is an idividual partition.
-This means further that every partition index is stricly unique and independent from all other items.
-Special character used within regular expressions (such as `+`, `*`, `^`) are officially not supported by the standard.
-The maximum size of each string has yet to be verified and may depend strongly on the implementation.
+Since this proposal splits the ROS name into namespace and base name, we'll consider how that might affect DDS partitions and DDS topic names respectively.
+When changing the namespace of a ROS name, i.e. everything but the base name, then this could be done by changing just the existing partition string.
+As partitions are part of QoS settings, it technically doesn't require a full restart of the publisher and subscription when changing the partitions field, however we cannot change the DDS topic name on the fly.
+With this in mind, we may want to always require a full restart of the publisher or subscription instance to avoid inconsistency in how these changes behave from the ROS user's perspective.
 
-DDS Partitions don't support a partition hierarchy.
-Each entry in the partition array is directly combined with the DDS topic and they are not sequentially combined.
-If a publisher operates on two partition entries, e.g. `foo` and `bar` with a base name of `baz`, the resulting topic will be twofold:
+### ROS Topic and Service Name Length Limit
 
-```
-- /foo/baz
-- /bar/baz
-```
+The length of the DDS topic must not exceed 255 characters.
+The actual length of a partition field may be limited to 255 characters, however this varies drastically depending on the vendor.
+RTI Connext does not allow a creation of a publisher/subscription with a partition length of 248 characters, whereas FastRTPS does not have any limitation in length.
+Please bare in mind, that the length of the partition gets further diminished due to the introduction of a ros specific prefix.
+The actual length of a ROS Topic, including the namespace hierarchy and the base name of the topic, may thus be varying in length as well.
+Yet, the base name token must not exceed the length of 255 characters as this is getting mapped direclty as the DDS topic.
 
-That implies, that DDS Partitions by default are not depicting any hierarchy.
+### Communicating with Non-ROS Topics
```

Comment:
I moved this section here, added notes about how the leading `/` of absolute topics might affect this and then added an examples and ideas section as well.